src/main/java/UtilsService.java

```diff
@@ -0,0 +1,120 @@
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@Slf4j
+public class UtilsService {
+
+    public boolean isValidJson(String responseText) {
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.readTree(responseText);
+            return true;
+        } catch (Exception e) {
+            log.error("Invalid JSON response: " + responseText);
+            return false;
+        }
+    }
+
+    public String preprocessGPTResponse(String responseText) {
+        if (responseText.startsWith("```json")) {
+            responseText = responseText.substring(7).trim();
```

Comment:
The substring operation assumes that the input is always valid. If responseText is shorter than 7 characters, it will throw an exception. (Severity: High)